worker-markup/src/main/java/com/hpe/caf/worker/markup/MarkupHeadersAndBody.java

```diff
@@ -154,6 +170,54 @@ else if (line.contains(": ")) {
     }
 
     /**
+     * Handle the correct markup of header fields which are surrounded with '*', i.e *Sent:* xxx.
+     *
+     * @param emailText the email text that is to be marked up.
+     * @param lines the emailText as separate lines.
+     * @param nattyParser the natty parser to detect the date language.
+     * @param headersElement the header element to add headers to.
+     * @return the email body index.
+     */
+    private int handleHeaderWithAsterisks(final String emailText, final String[] lines, final Parser nattyParser,
+                                          final Element headersElement, int bodyIndex)
+    {
+        // Check to see if the the from field is split onto two lines.
+        final Matcher splitFromFieldMatcher = fromFieldSplitOntoTwoLines.matcher(emailText);
+        final List<String> asteriskHeadersAlreadyMarkedUp = new ArrayList<>();
+
+        if (splitFromFieldMatcher.find()) {
+            for (final String line : lines) {
+                // For a line which matches the format "*From:*example.emailaddress.com",
+                // concat this and the second part of the 'From' field "[mailto:example.emailaddress.com] *On Behalf Of *Bloggs, Joe"
+                // So that the whole value is marked up with <From>
+                if (line.equals(splitFromFieldMatcher.group("FirstPartFromField"))) {
+                    asteriskHeadersAlreadyMarkedUp.add(line);
+                    bodyIndex++;
+                    final String fullFromFieldValue = line + "\n" + splitFromFieldMatcher.group("SecondPartFromField");
+                    addStandardisedHeader(nattyParser, headersElement, lines, fullFromFieldValue, ":\\*");
+                } // Once there is a match for the second part of the <From> value just increase the bodyIndex
+                else if (line.equals(splitFromFieldMatcher.group("SecondPartFromField"))) {
+                    asteriskHeadersAlreadyMarkedUp.add(line);
+                    bodyIndex++;
+                }
```

Comment:
Removing the `break;` doesn't remove the assumption that From is the first header.